File: C_Programming/Practise_Programs/PP_160.c
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_160
//
//  Description       : This program displays the decimal value of
//                      powers of 2 from the 1st bit to the 32nd bit
//                      using the left shift operator.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 19/08/2026
//
//  Time Complexity   : O(n)
//  Space Complexity  : O(1)
//
//////////////////////////////////////////////////////////////////

#include<stdio.h>

typedef unsigned int UNIT;

int main()
{
    UNIT iMask = 1;
    int iCnt = 0;

    for(iCnt = 1; iCnt <= 32; iCnt++)
    {
        printf("%d : %up\n", iCnt, iMask);  
        iMask = iMask << 1;
    }   
    
    return 0;
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input :
//      No input required.
//
//  Sample Output :
//
//      1 : 1p
//      2 : 2p
//      3 : 4p
//      4 : 8p
//      5 : 16p
//      ...
//      31 : 2147483648p
//      32 : 0p
//
//////////////////////////////////////////////////////////////////

File: Java_Programming/PP_160.java
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_160
//
//  Description       : This program accepts a number from the
//                      user and displays lowercase alphabets
//                      at odd positions and a star at even
//                      positions.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 03/09/2026
//
//  Time Complexity   : O(n)
//  Space Complexity  : O(1)
//
//////////////////////////////////////////////////////////////////

import java.util.Scanner;


class PP_160
{
    public static void Display(int iNo)
    {
        int iCnt = 0;
        char ch = '\0';

        for(iCnt = 1, ch = 'a'; iCnt <= iNo; iCnt++)
        {
            if(iCnt % 2 == 0)
            {
                System.out.print("*\t");
            }
            else
            {
                System.out.print(ch + "\t");
                ch++;
            }            
        }
        System.out.println();
    }
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        int iValue = 0;

        System.out.println("Enter the elements : ");
        iValue = sobj.nextInt();

        Display(iValue);
    }
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input :
//      Enter the elements : 5
//
//  Sample Output :
//
//      a	*	b	*	c
//
//////////////////////////////////////////////////////////////////